### DISCUSSION TITLE
Stacking results without altering the previous, extending the speclib

### DISCUSSION THREAD

**Discussion by Sacesh** [vdemichev/DiaNN#1112] [Q&A]
Hello community,

I am trying to create a speclib from Fasta and resulting pgmatrix using DIA-NN without applying default normalization or correction steps. The pgmatrix is to be "extended" in subsequent runs.

In a subsequent run, I want to use the previously generated speclib and .quant files along with new raw data. 

The goal is to produce a pgmatrix that integrates results from the new raw files without altering the quantification of the previous raw files, being the reason for not using the diann default normalization or correction steps.

Ideally, there should be settings to "extend" the speclib (maybe again together with the fasta) with the addition of new protein groups from new samples, with NaN values assigned to previous samples if they were not detected previously. I wouldnt want to make a full new run, as i want to keep previous quantifications (without adding them manually).

I already tried multiple permutations of settings, with errors because p.E. of changed settings to previous run or not having pgmatrices or tsv libs being generated at all. Also i'm not entirely sure which spectral lib file is the correct one to link.
Setting i tested last:
--relaxed-prot-inf
--no-norm
_--no-maxlfq_

It would be much appreciated if someone could share their insights or walk me through a valid approach here, as test runs take quite some time.

Best,
Nik

**Comment by vdemichev** [comment]
Hi Nik,

Please try the incremental data processing as described in https://github.com/vdemichev/DiaNN?tab=readme-ov-file#frequently-asked-questions. This will not work with pg_matrix (due to global q-value filtering of matrices), but will work fine with the main report.

> In a subsequent run, I want to use the previously generated speclib and .quant files along with new raw data.

Please only use .parquet (1.9.1) or .tsv libraries (1.9) generated by DIA-NN from DIA data, not .speclib. The .speclib is currently only produced for Skyline integration.

Best,
Vadim
